### PR TITLE
Option to use API Gateway in commercial partition.

### PR DESCRIPTION
**Pull request by chriswvu**
Feature update to support deploying API Gateway in commercial partition.

### CODE REVIEW

File: data.tf [hunk 1]
```diff
@@ -1,7 +1,8 @@
 locals {
+  api_gw_enabled = local.is_govcloud || var.api_gw_enabled == true ? toset(["enabled"]) : []
```
Comment (lorengordon) [inline data.tf]:
```suggestion
  api_gw_enabled = local.is_govcloud || var.api_gw_enabled ? toset(["enabled"]) : []
```

File: data.tf [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
   is_govcloud     = data.aws_partition.current.partition == "aws-us-gov"
   each_govcloud   = local.is_govcloud ? toset(["enabled"]) : []
```
Comment (lorengordon) [inline data.tf]: Is this still used at all? Maybe delete?

File: data.tf [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
-  each_commercial = local.is_govcloud ? [] : toset(["enabled"])
+  each_commercial = local.is_govcloud || var.api_gw_enabled == true ? [] : toset(["enabled"])
```
Comment (lorengordon) [inline data.tf]: Maybe change to `function_url_enabled`?

```suggestion
  function_url_enabled = local.is_govcloud || var.api_gw_enabled ? [] : toset(["enabled"])
```

File: outputs.tf [hunk 4]
```diff
@@ -5,7 +5,7 @@ output "courier_function_arn" {
 
 output "courier_url" {
   description = "The HTTP URL endpoint for the courier"
-  value       = local.is_govcloud ? aws_apigatewayv2_stage.this["enabled"].invoke_url : aws_lambda_function_url.courier["enabled"].function_url
+  value       = local.is_govcloud || var.api_gw_enabled == true ? aws_apigatewayv2_stage.this["enabled"].invoke_url : aws_lambda_function_url.courier["enabled"].function_url
```
Comment (lorengordon) [inline outputs.tf]:
```suggestion
  value       = local.api_gateway_enabled ? aws_apigatewayv2_stage.this["enabled"].invoke_url : aws_lambda_function_url.courier["enabled"].function_url
```

File: variables.tf [hunk 5]
```diff
@@ -1,3 +1,9 @@
+variable "api_gw_enabled" {
+  default     = false
+  description = "Install api gateway in commercial"
+  type        = bool
+}
```
Comment (lorengordon) [inline variables.tf]: Add `nullable = false` to avoid case requiring testing for null...

```suggestion
variable "api_gw_enabled" {
  default     = false
  description = "Install api gateway in commercial"
  nullable    = false
  type        = bool
}
```

File: outputs.tf [hunk 6]
```diff
@@ -5,7 +5,7 @@ output "courier_function_arn" {
 
 output "courier_url" {
   description = "The HTTP URL endpoint for the courier"
-  value       = local.is_govcloud ? aws_apigatewayv2_stage.this["enabled"].invoke_url : aws_lambda_function_url.courier["enabled"].function_url
+  value       = local.is_govcloud || var.api_gw_enabled ? aws_apigatewayv2_stage.this["enabled"].invoke_url : aws_lambda_function_url.courier["enabled"].function_url
```
Comment (lorengordon) [inline outputs.tf]: Changing this condition to use the new `local.api_gw_enabled` should allow us to get rid of `local.each_govcloud` entirely...

```suggestion
  value       = local.api_gw_enabled ? aws_apigatewayv2_stage.this["enabled"].invoke_url : aws_lambda_function_url.courier["enabled"].function_url
```

**Review by jubranNassar — APPROVED** [review]
